core/imt/inc/ROOT/TExecutor.hxx

```diff
@@ -0,0 +1,494 @@
+// @(#)root/thread:$Id$
+// Author: Xavier Valls September 2020
+
+/*************************************************************************
+ * Copyright (C) 1995-2006, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+#ifndef ROOT_TExecutor
+#define ROOT_TExecutor
+
+#include <ROOT/RMakeUnique.hxx>
+#include "ROOT/TExecutorCRTP.hxx"
```

Comment:
Why is this needed?

Reply:
@Axel-Naumann Do you mean the #define? I am not sure if they are still necessary or not